src/server/pps/server/pipeline_controller.go

```diff
@@ -79,6 +73,7 @@ type pipelineController struct {
 
 	bumpChan chan time.Time
 	pcMgr    *pcManager
+	kd       *kubeDriver
```

Comment:
seems like we can lose the k8s-relevant fields `namespace`, `etcdPrefix`) from the pipelineController

Reply:
right!